control_plane/src/attachment_service.rs

```diff
@@ -101,4 +111,29 @@ impl AttachmentService {
         let response = response.json::<AttachHookResponse>()?;
         Ok(response.gen)
     }
+
+    pub fn inspect(&self, tenant_id: TenantId) -> anyhow::Result<Option<(u32, NodeId)>> {
+        use hyper::StatusCode;
+
+        let url = self
+            .env
+            .control_plane_api
+            .clone()
+            .unwrap()
+            .join("inspect")
+            .unwrap();
+        let client = reqwest::blocking::ClientBuilder::new()
+            .build()
+            .expect("Failed to construct http client");
```

Comment:
It seems all methods build their own client; this is a general reqwest-usage antipattern but I guess in all cases only one of these is executed at a process? It might be the that the migration will cause reation of multiple clients.

Not a huge problem, I just must note this.

Reply:
Cleaned up in 7f8566a7f